dotCMS/src/main/java/com/dotmarketing/image/filter/CropImageFilter.java

```diff
@@ -1,61 +1,103 @@
 package com.dotmarketing.image.filter;
 
+import java.awt.Dimension;
+import java.awt.Point;
 import java.awt.image.BufferedImage;
 import java.io.File;
 import java.io.IOException;
 import java.util.Map;
-
+import java.util.Optional;
 import javax.imageio.ImageIO;
-
+import com.dotmarketing.image.focalpoint.FocalPoint;
+import com.dotmarketing.image.focalpoint.FocalPointAPIImpl;
 import com.dotmarketing.util.Logger;
 
+/**
+ * Crop a image focusing in a point
+ */
 public class CropImageFilter extends ImageFilter {
 
+	public static final String X_PARAM_KEY = "x";
+	public static final String Y_PARAM_KEY = "y";
+	public static final String WIDTH_PARAM_KEY  = "w";
+	public static final String HEIGHT_PARAM_KEY = "h";
+
 	public String[] getAcceptedParameters(){
 		return  new String[] {
 				"x (int) for left of crop",
 				"y (int) for top of crop",
 				"w (int) for width of crop",
-				"h (int) for height of crop"
+				"h (int) for height of crop",
+				"fp (int,int) the focal point of the crop"
 		};
 	}
 	
 	
 	
-	public File runFilter(File file,  Map<String, String[]> parameters) {
-		int x = parameters.get(getPrefix() + "x") != null ? Integer.parseInt(parameters.get(getPrefix() + "x")[0]) : 0;
-		int y = parameters.get(getPrefix() + "y") != null ? Integer.parseInt(parameters.get(getPrefix() + "y")[0]) : 0;
-		int w = parameters.get(getPrefix() + "w") != null ? Integer.parseInt(parameters.get(getPrefix() + "w")[0]) : 0;
-		int h = parameters.get(getPrefix() + "h") != null ? Integer.parseInt(parameters.get(getPrefix() + "h")[0]) : 0;
-		if (w == 0 || h == 0) {
-			return file;
-		}
-		
-		File resultFile = getResultsFile(file, parameters);
+	public File runFilter(final File file,  final Map<String, String[]> parameters) {
+		int x = parameters.get(getPrefix() + X_PARAM_KEY) != null ? Integer.parseInt(parameters.get(getPrefix() + X_PARAM_KEY)[0]) : 0;
+		int y = parameters.get(getPrefix() + Y_PARAM_KEY) != null ? Integer.parseInt(parameters.get(getPrefix() + Y_PARAM_KEY)[0]) : 0;
+		final float widthInput  = parameters.get(getPrefix() + WIDTH_PARAM_KEY)  != null ? Float.parseFloat(parameters.get(getPrefix()  + WIDTH_PARAM_KEY)[0]) : 0f;
+		final float heightInput = parameters.get(getPrefix() + HEIGHT_PARAM_KEY) != null ? Float.parseFloat(parameters.get(getPrefix() + HEIGHT_PARAM_KEY)[0]) : 0f;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4761083)